purple/core.py

```diff
@@ -106,6 +133,15 @@ def toggle_lock(self, keycodes):
             else:
                 self._lock_key_buffer.append(keycode)
 
+    def toggle_hold(self, keycodes):
+        for keycode in keycodes:
+            if keycode in self._held_keys:
+                self._held_keys.remove(keycode)
+                self._hid_keyboard.press(keycode)
```

Comment:
the `press` and `release` are flipped here and in the `else`, if removing a keycode you need to `release` it, and if adding one you should `press` the key